src/main/java/org/tugraz/sysds/pythonapi/pythonDMLScript.java

```diff
@@ -0,0 +1,36 @@
+package org.tugraz.sysds.pythonapi;
+
+import org.tugraz.sysds.api.jmlc.Connection;
+import py4j.GatewayServer;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+
+public class pythonDMLScript {
+	private Connection _connection;
+	
+	public static void main(String[] args) {
+		GatewayServer gatewayServer = new GatewayServer(new pythonDMLScript());
+		gatewayServer.start();
+		System.out.println("Gateway Server Started");
+		// copied from: https://stackoverflow.com/questions/43282447/py4j-launch-gateway-not-connecting-properly
+		if (args.length == 1 && args[0].equals("--die-on-exit")) {
+			try {
+				BufferedReader stdin = new BufferedReader(new InputStreamReader(System.in));
+				stdin.readLine();
+				System.exit(0);
```

Comment:
If you split this up, such that you make a public function that returns a boolean for if the program fails or passes, then the testing is easier. If you use System exit 0 and 1 then it also will exit the testing.

So i would suggest moving this main methods logic to another function, and then do the System exit here in the main function, while returning a Boolean from the new function.